src/solvers/dgsem_tree/dg_1d.jl

```diff
@@ -76,6 +76,9 @@ function rhs!(du, u, t,
   # Reset du
   @trixi_timeit timer() "reset ∂u/∂t" reset_du!(du, dg, cache)
 
+  # Apply waterheight and velocity cut off for ShallowWaterEquations1D
+  apply_thresholds!(u, equations, dg, cache)
```

Comment:
This thresholding needs to happen for the wet/dry solver. But since this is so ShallowWater specific would we maybe need a special `rhs!` function that dispatches on `equations`? My main question would be on performance versus readability. What do you think @ranocha and @sloede ?